File: easy/28. Find the Index of the First Occurrence in a String/28. Find the Index of the First Occurrence in a String.ts
function strStr(haystack: string, needle: string): number {
    // let index = 0
    // const needle_length = needle.length
    // while (index < haystack.length) {
    //     const right = index + needle_length
    //     if (haystack.slice(index, right) === needle)
    //         return index
    //     index += 1
    // }
    // return -1

    return haystack.indexOf(needle)
};

File: easy/28. Find the Index of the First Occurrence in a String/28. Find the Index of the First Occurrence in a String.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        index = 0
        needle_length = len(needle)

        while index < len(haystack):
            right = index + needle_length

            if haystack[index:right] == needle:
                return index
            index += needle_length

        return -1


solution = Solution()

print(solution.strStr("sadbutsad", "sad"))
print(solution.strStr("leetcode", "leeto"))
